internal/gitutil/git.go

```diff
@@ -69,7 +70,27 @@ func EnsureUpdated(uri, destinationPath string) error {
 	return updateAndCleanUntracked(destinationPath)
 }
 
-func exec(pwd string, args ...string) error {
+// ListModifiedFiles will fetch origin and list modified files
+func ListModifiedFiles(uri, destinationPath string) ([]string, error) {
+	if _, err := exec(destinationPath, "fetch", "-v"); err != nil {
+		return []string{}, errors.Wrapf(err, "fetch index at %q failed", destinationPath)
+	}
+
+	output, err := exec(destinationPath, "diff", "--name-only", "@{upstream}", "--", ".")
```

Comment:
Sorry if I come up with those ideas bit by bit. But you can also use `git diff --name-status` which will add a mark each change as modified/added/deleted...

You can use that to further simplify the logic you need to build the report. I suggest that you play around with a toy git repo to see how this works. Try to offload as much as possible to git :)